File: 조준장/level1/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    
    for target in targets:
        cnt = 0
        for t in target:
            key_idx = []
            for k in keymap:
                if t in k:
                    key_idx.append(k.index(t))
            if not key_idx:
                cnt = -1
                break
            cnt += min(key_idx)+1
        answer.append(cnt)
    
    
    return answer

File: 이훈일/Programmers/Level1/대충 만든 자판.java
class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];

        for(int i = 0; i < targets.length; i++) {
            String[] targetArr = targets[i].split("");
            int count = 0;
            for(String s : targetArr) {
                int index = -1;
                for(String key : keymap) {
                    if(key.contains(s)) {
                        if(index < 0){
                            index = key.indexOf(s) + 1;
                        }else{
                            if(index > key.indexOf(s)) {
                                index = key.indexOf(s) + 1;
                            }
                        }
                    }
                }
                if(count != -1) {
                    if(index == -1){
                        count = -1;
                    } else {
                        count += index;
                    }
                }
            }
            answer[i] = count;
        }

        return answer;
    }
}
